xbmc/video/tags/ISetInfoTagLoader.h

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright (C) 2017-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "InfoScanner.h"
+#include "addons/Scraper.h"
+
+#include <string>
+#include <utility>
+
+class CFileItem;
+class CSetInfoTag;
+class EmbeddedArt;
+
+namespace VIDEO
+{
+
+//! \brief Base class for set tag loaders.
+class ISetInfoTagLoader
+{
+public:
+  //! \brief Constructor
+  //! \param item The item to load info for
+  //! \param info Scraper info
+  //! \param llokInFolder True to look in folder holding file
+  ISetInfoTagLoader(const CFileItem& item, ADDON::ScraperPtr info, bool lookInFolder = false)
+    : m_item(item), m_info(std::move(info))
+  {
+  }
+  virtual ~ISetInfoTagLoader() = default;
+
+  //! \brief Returns true if we have info to provide.
+  virtual bool HasInfo() const = 0;
+
+  //! \brief Load tag from file.
+  //! \brief tag Tag to load info into
+  //! \brief prioritise True to prioritise data over existing data in tag
+  //! \returns True if tag was read, false otherwise
+  virtual CInfoScanner::INFO_TYPE Load(CSetInfoTag& tag,
+                                       bool prioritise,
+                                       std::vector<EmbeddedArt>* art = nullptr) = 0;
```

Comment:
I would not use raw pointers. Or has it to be like this? Could this be a `std::unique_ptr<std::vector<...`? Or you get just rid of the default parameter?

Reply:
I tied myself in knots over template classes as I've never used them before. I assumed that the declaration of Load had to be the same as that in `IVideoInfoTagLoader` - but I'll try without it, as it's not used.

Reply:
It seems to be ok, thanks.